src/home/pam_systemd_home.c

```diff
@@ -727,7 +727,7 @@ _public_ PAM_EXTERN int pam_sm_open_session(
 
         r = acquire_home(handle, /* please_authenticate = */ false, suspend_please, debug);
         if (r == PAM_USER_UNKNOWN) /* Not managed by us? Don't complain. */
-                return PAM_SUCCESS;
+                goto success; /* Need to free the bus resource, as acquire_home() takes a reference. */
```

Comment:
so this looks good, but I think we should also disconnect on failure explicitly. That's because we actually hook all our modules in via a PAM stack line that is prefixed with "-", we tell it to ignore failures.

Reply:
We probably should yes, however it's better to do that separately. This PR should fix the immediate visible issues, then we can follow-up with that and more - I have all the changes for the random nonce for sd-bus for example, which I'll send separately